### PR TITLE
Apply an additional condition if -XX:-CRIUSecProvider is specified

### PR DESCRIPTION
**Pull request by WilburZjh**
Back-ported from https://github.com/ibmruntimes/openj9-openjdk-jdk/pull/689

### CODE REVIEW

**Comment by WilburZjh** [comment]
@jasonkatonica FYI

File: src/java.base/share/classes/java/security/Security.java [hunk 1]
```diff
@@ -142,9 +142,13 @@ private static void initialize() {
...
         /*[IF CRIU_SUPPORT]*/
-        // Check if CRIU checkpoint mode is enabled, if it is then reconfigure the security providers.
-        if (InternalCRIUSupport.isCheckpointAllowed()) {
+        // If -XX:-CRIUSecProvider is specified, we don't need to configure the CRIUSec provider.
+        if (InternalCRIUSupport.isCheckpointAllowed() && InternalCRIUSupport.enableCRIUSecProvider()) {
             CRIUConfigurator.setCRIUSecMode(props);
+            // Check if `-XX:+EnableFullSecurityEnv` is enabled, if it is then reconfigure the security providers.
+            if (InternalCRIUSupport.isEnableFullSecurityEnvInCRIU()) {
+                CRIUConfigurator.setCRIURestoreMode();
+            }
```
Comment (keithc-ca) [inline src/java.base/share/classes/java/security/Security.java:151]: This should be the same as in later streams (also `isEnableFullSecurityEnvInCRIU` doesn't exist).

**Review by keithc-ca — CHANGES_REQUESTED** [review]
Please update the commit message; it should be "written in the imperative mood".
See https://github.com/eclipse-openj9/openj9/blob/master/CONTRIBUTING.md#commit-guidelines.

**Comment by keithc-ca** [comment]
I see you changed the title of this PR: The commits should be squashed and the commit message should be similar.

**Comment by WilburZjh** [comment]
> I see you changed the title of this PR: The commits should be squashed and the commit message should be similar.

Yes, once the build is finished, i will update them all.

**Comment by WilburZjh** [comment]
Build pass: https://hyc-runtimes-jenkins.swg-devops.com/view/OpenJ9%20-%20Personal/job/Pipeline-Build-Test-Personal/19781/

Commits are squashed

**Comment by keithc-ca** [comment]
Jenkins test sanity plinux jdk17

**Review by keithc-ca — APPROVED** [review]

**Pull request merged**